### ISSUE TITLE
NoMethodError in Admin::Users#index

### ISSUE THREAD

**Issue by EricRouach** [activeadmin/activeadmin#5398]
Hi everyone, 

I recently stumble upon a new bug when opening the tab User in my dashboard.
The tab is supposed to reference all the users of my app, and it was working until today.

Now, when I try to open this tab (all the others are working), I get an "undefined method `notifications_id_eq' for Ransack::Search<class: User, base: Grouping <combinator: and>>:Ransack::Search"
(located in "app/views/active_admin/resource/index.html.arb"). 

I am using Rails 5.1.4 and ruby 2.4.3.

PS: I haven't done any recent changes to the User table in my DB 

Cheers,
Eric Rouach

**Comment by varyonic** [comment]
Try posting a stack trace directly or linking to pastebin.

**Comment by EricRouach** [comment]
Hello, 
Sorry for the late answer! 
I still have the issue, so I am commenting it here (as I cannot re-open the issue myself)

Here is the error I get when i try to access the user tab in the admin panel:
<img width="570" alt="screen shot 2018-04-24 at 11 55 30" src="https://user-images.githubusercontent.com/28184166/39180258-a6b41cf0-47b6-11e8-8e3c-a83a60d4ca37.png">

Here is my schema (db, postgresql):
<img width="484" alt="screen shot 2018-04-24 at 11 55 14" src="https://user-images.githubusercontent.com/28184166/39180281-b6b23498-47b6-11e8-8700-cc2076bc4f17.png">

And here is the configuration of the user tab for the active admin panel:
<img width="841" alt="screen shot 2018-04-24 at 11 55 07" src="https://user-images.githubusercontent.com/28184166/39180302-c93076e8-47b6-11e8-8d4f-ae1959ff2c90.png">

Many thanks,
Eric Rouach

**Comment by Fivell** [comment]
@EricRouach can you show your User model, seems there is some error declaring associations

**Comment by EricRouach** [comment]
hey @Fivell, 
You were right, I still had an old association declared. By removing it, I resolved the first error, but another is thrown right away. 

<img width="790" alt="screen shot 2018-04-27 at 17 33 42" src="https://user-images.githubusercontent.com/28184166/39371156-27f288f6-4a41-11e8-9966-d18f04968e4e.png">

I think I might have found the cause: I am using a notification table with composite keys (user_id and type_of_notifications). 

<img width="791" alt="screen shot 2018-04-27 at 17 34 27" src="https://user-images.githubusercontent.com/28184166/39371192-408fc180-4a41-11e8-80d4-a197eea88f15.png">

Does Active Admin support this?  I did not create an admin panel for the notification table though..
